binary/private/binary_roche_deformation.f90

```diff
@@ -0,0 +1,357 @@
+!   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+!   See the GNU Library General Public License for more details.
+!
+!   You should have received a copy of the GNU Library General Public License
+!   along with this software; if not, write to the Free Software
+!   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+!
+! ***********************************************************************
+
+module binary_roche_deformation
+
+   ! computes the stellar structure correction factors fp/ft and the specific
+   ! moment of inertia i_rot assuming shellularity in the Roche potential of
+   ! a binary star. Follows Fabry, Marchant and Sana, 2022, A&A 661, A123.
+
+   use ieee_arithmetic, only: ieee_is_nan
+   use interp_2d_lib_db
+   use auto_diff
+   use star_def
+   use binary_def
+   use binary_utils, only : eval_rlobe
+
+
+   implicit none
+
+   real(dp), parameter :: nudge = 1d-4
+   real(dp), pointer :: xvals(:), yvals(:), yvals_gtr_than_1(:), fpfunc1d(:), ftfunc1d(:), &
+         irotfunc1d(:), otherrfunc1d(:), afunc1d(:)
+   logical :: inter_ok = .false., dbg = .true.
+   integer :: num_xpts, num_ypts, num_ypts_gtr_than_1
+
+contains
+
+   subroutine build_roche_interpolators
+      use const_def, only: mesa_data_dir
+      real(dp) :: xtest, ytest, testval
+      integer :: ierr
+      character(len=strlen) :: upstairs
+      include 'formats'
+      if (.not. inter_ok) then
+         upstairs = trim(mesa_data_dir) // '/roche_data/'  ! where fp/ft data lives
+         if (dbg) then
+            write(*, 1) 'starting interpolator setup'
+         end if
+         call setup_interpolator(trim(upstairs) // 'fp_data.txt', xvals, num_xpts, yvals, &
+               num_ypts, fpfunc1d, ierr)
+         call setup_interpolator(trim(upstairs) // 'ft_data.txt', xvals, num_xpts, yvals, &
+               num_ypts, ftfunc1d, ierr)
+         call setup_interpolator(trim(upstairs) // 'irot_data.txt', xvals, num_xpts, &
+               yvals, num_ypts, irotfunc1d, ierr)
+         if (dbg) then
+            xtest = -0.5
+            ytest = 1.35
+            ! test fp interpolator
+            write(*, 11) 'grid size', num_xpts, num_ypts
+            write(*, 1) 'setup interpolators succesful,'
+
+            call interp_evbipm_db(xtest, ytest, xvals, num_xpts, yvals, num_ypts,&
+                  fpfunc1d, num_xpts, testval, ierr)
+            write(*, 1) 'fp   test gave should be close to 0.6', testval
+            call interp_evbipm_db(xtest, ytest, xvals, num_xpts, yvals, num_ypts,&
+                  ftfunc1d, num_xpts, testval, ierr)
+            write(*, 1) 'ft   test gave should be close to 0.8', testval
+            call interp_evbipm_db(xtest, ytest, xvals, num_xpts, yvals, num_ypts,&
+                  irotfunc1d, num_xpts, testval, ierr)
+            write(*, 1) 'irot test gave should be close to 0.4', testval
+            inter_ok = .true.
+         end if
```

Comment:
I won't have a chance to look at this closely for a bit but do we want to `inter_ok` to `.true.` *inside* the debug block? Should it not be set to `.true.` in normal operation? I'd expect
```suggestion
         end if
         inter_ok = .true.
```

Reply:
you're absolutely right!